### PR TITLE
AGraph integration with radiff2

### PR DESCRIPTION
**Pull request by deepakchethan**
Interactive graph (radiff2 -g main -m i false true):

Screen Shot 2019-08-18 at 5 43 27 PM

### CODE REVIEW

File: libr/main/radiff2.c [hunk 1]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+        static_str = r_cons_get_buffer ();
+        retstr = strdup (static_str? static_str: "");
+        r_cons_pop ();
+        r_cons_echo (NULL);
+        r_cons_singleton ()->is_html = tmp_html;
+        return retstr;
+}
+
+static void __generate_graph (RCore *c, ut64 off) {
+        r_return_if_fail(c);
```
Comment (XVilka) [inline libr/main/radiff2.c]: space between

File: libr/main/radiff2.c [hunk 2]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+        char *eol, *ptr = get_graph_commands(c, off);
+        r_cons_break_push (NULL, NULL);
+        if (ptr) {
+                for (;;) {
+                        if (r_cons_is_breaked ()) {
+                                break;
+                        }
+                        eol = strchr (ptr, '\n');
```
Comment (radare) [inline libr/main/radiff2.c]: define  and assign  here

File: libr/main/radiff2.c [hunk 3]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+        retstr = strdup (static_str? static_str: "");
+        r_cons_pop ();
+        r_cons_echo (NULL);
+        r_cons_singleton ()->is_html = tmp_html;
+        return retstr;
+}
+
+static void __generate_graph (RCore *c, ut64 off) {
+        r_return_if_fail(c);
+        char *eol, *ptr = get_graph_commands(c, off);
```
Comment (radare) [inline libr/main/radiff2.c]: missing space

File: libr/main/radiff2.c [hunk 4]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+                        if (r_cons_is_breaked ()) {
+                                break;
+                        }
+                        eol = strchr (ptr, '\n');
+                        if (eol) {
+                                *eol = '\0';
+                        }
+                        if (*ptr) {
+                                char *p = strdup (ptr);
+                                r_core_cmd0 (c, p);
```
Comment (radare) [inline libr/main/radiff2.c]: nullchk p

File: libr/main/radiff2.c [hunk 5]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+                break;
+        case GRAPH_JSON_MODE:
+                r_core_anal_graph (c, off, opts | R_CORE_ANAL_JSON);
+                break;
+        case GRAPH_JSON_DIS_MODE:
+                r_core_anal_graph (c, off, opts | R_CORE_ANAL_JSON | R_CORE_ANAL_JSON_FORMAT_DISASM);
+                break;
+        case GRAPH_DEFAULT_MODE:
+        default:
+                __generate_graph(c, off);
```
Comment (radare) [inline libr/main/radiff2.c]: missing space

File: libr/main/radiff2.c [hunk 6]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
+                r_core_cmd0 (c, "agg");
+        break;
+        }
+        r_cons_reset_colors();
```
Comment (radare) [inline libr/main/radiff2.c]: missing space

File: libr/main/radiff2.c [hunk 7]
```diff
@@ -886,15 +1007,14 @@ R_API int r_main_radiff2(int argc, char **argv) {
 				r_core_anal_fcn (c2, r_num_math (c2->num, second),
 					UT64_MAX, R_ANAL_REF_TYPE_NULL, depth);
 				r_core_gdiff (c, c2);
-				r_core_anal_graph (c, off, R_CORE_ANAL_GRAPHBODY | R_CORE_ANAL_GRAPHDIFF);
+                                __print_diff_graph(c, off, gmode);
```
Comment (radare) [inline libr/main/radiff2.c]: use tabs not spaces

File: libr/main/radiff2.c [hunk 8]
```diff
@@ -886,15 +1007,14 @@ R_API int r_main_radiff2(int argc, char **argv) {
...
 			} else {
 				r_core_anal_fcn (c, r_num_math (c->num, words),
 					UT64_MAX, R_ANAL_REF_TYPE_NULL, depth);
 				r_core_anal_fcn (c2, r_num_math (c2->num, words),
 					UT64_MAX, R_ANAL_REF_TYPE_NULL, depth);
 				r_core_gdiff (c, c2);
-				r_core_anal_graph (c, r_num_math (c->num, addr),
-					R_CORE_ANAL_GRAPHBODY | R_CORE_ANAL_GRAPHDIFF);
+                                __print_diff_graph(c, r_num_math (c->num, addr), gmode);
```
Comment (radare) [inline libr/main/radiff2.c]: use tabs not spaces and add missing space before (

File: libr/main/radiff2.c [hunk 9]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
 
+static char *get_graph_commands (RCore *c, ut64 off) {
+        char *retstr;
+        const char *static_str = NULL;
+        int tmp_html = r_cons_singleton ()->is_html;
+        r_cons_singleton ()->is_html = 0;
+        r_cons_push ();
+        r_core_anal_graph (c, off, R_CORE_ANAL_GRAPHBODY | R_CORE_ANAL_GRAPHDIFF |  R_CORE_ANAL_STAR);
+        static_str = r_cons_get_buffer ();
+        retstr = strdup (static_str? static_str: "");
```
Comment (radare) [inline libr/main/radiff2.c]: use tabs to indent!

File: libr/main/radiff2.c [hunk 10]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
...
 }
 
+static char *get_graph_commands (RCore *c, ut64 off) {
+        char *retstr;
+        const char *static_str = NULL;
+        int tmp_html = r_cons_singleton ()->is_html;
+        r_cons_singleton ()->is_html = 0;
+        r_cons_push ();
+        r_core_anal_graph (c, off, R_CORE_ANAL_GRAPHBODY | R_CORE_ANAL_GRAPHDIFF |  R_CORE_ANAL_STAR);
+        static_str = r_cons_get_buffer ();
```
Comment (radare) [inline libr/main/radiff2.c]: define and assign in the same line

File: libr/main/radiff2.c [hunk 11]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
 	return buf;
 }
 
+static char *get_graph_commands (RCore *c, ut64 off) {
+        char *retstr;
+        const char *static_str = NULL;
+        int tmp_html = r_cons_singleton ()->is_html;
+        r_cons_singleton ()->is_html = 0;
```
Comment (radare) [inline libr/main/radiff2.c]: use false, not 0, this is bool, or it shuold...

File: libr/main/radiff2.c [hunk 12]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
 	return buf;
 }
 
+static char *get_graph_commands (RCore *c, ut64 off) {
+        char *retstr;
+        const char *static_str = NULL;
+        int tmp_html = r_cons_singleton ()->is_html;
```
Comment (radare) [inline libr/main/radiff2.c]: this must be bool

File: libr/main/radiff2.c [hunk 13]
```diff
@@ -696,6 +719,89 @@ static ut8 *get_strings(RCore *c, int *len) {
 	return buf;
 }
 
+static char *get_graph_commands (RCore *c, ut64 off) {
```
Comment (radare) [inline libr/main/radiff2.c]: function definitions dont have a space before (

File: libr/main/radiff2.c (hunk 1 above)
Comment (deepakchethan) [reply libr/main/radiff2.c]: Added

File: libr/main/radiff2.c (hunk 11 above)
Comment (deepakchethan) [reply libr/main/radiff2.c]: It is actually an int sir, but I changed it to false

**Comment by XVilka** [comment]
Good. Will merge once is green

**Comment by deepakchethan** [comment]
I fixed the issues regarding this pr, CI is failing because of some other issues I believe

**Pull request merged**